api/test/helpers/keystone.go

```diff
@@ -0,0 +1,252 @@
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package helpers
+
+import (
+	"context"
+	"time"
+
+	"github.com/go-logr/logr"
+	"github.com/google/uuid"
+	"github.com/onsi/gomega"
+	keystonev1 "github.com/openstack-k8s-operators/keystone-operator/api/v1beta1"
+	"github.com/openstack-k8s-operators/lib-common/modules/common/condition"
+	"github.com/openstack-k8s-operators/lib-common/modules/test/apis"
+	k8s_errors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	base "github.com/openstack-k8s-operators/lib-common/modules/common/test/helpers"
+)
+
+// TestHelper is a collection of helpers for testing operators. It extends the
+// generic TestHelper from modules/test.
+type TestHelper struct {
+	*base.TestHelper
+}
+
+// NewTestHelper returns a TestHelper
+func NewTestHelper(
+	ctx context.Context,
+	k8sClient client.Client,
+	timeout time.Duration,
+	interval time.Duration,
+	logger logr.Logger,
+) *TestHelper {
+	helper := &TestHelper{}
+	helper.TestHelper = base.NewTestHelper(ctx, k8sClient, timeout, interval, logger)
+	return helper
+}
+
+// CreateKeystoneAPI creates a new KeystoneAPI instance with the specified namespace in the Kubernetes cluster.
+//
+// Example usage:
+//
+//	keystoneAPI := th.CreateKeystoneAPI(namespace)
+//	DeferCleanup(th.DeleteKeystoneAPI, keystoneAPI)
+func (th *TestHelper) CreateKeystoneAPI(namespace string) types.NamespacedName {
+	keystone := &keystonev1.KeystoneAPI{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "keystone.openstack.org/v1beta1",
+			Kind:       "KeystoneAPI",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "keystone-" + uuid.New().String(),
+			Namespace: namespace,
+		},
+		Spec: keystonev1.KeystoneAPISpec{},
+	}
+
+	gomega.Expect(th.K8sClient.Create(th.Ctx, keystone.DeepCopy())).Should(gomega.Succeed())
+	name := types.NamespacedName{Namespace: namespace, Name: keystone.Name}
+
+	// the Status field needs to be written via a separate client
+	keystone = th.GetKeystoneAPI(name)
+	keystone.Status = keystonev1.KeystoneAPIStatus{
+		APIEndpoints: map[string]string{
+			"public":   "http://keystone-public-openstack.testing",
+			"internal": "http://keystone-internal.openstack.svc:5000",
+		},
+	}
+	gomega.Expect(th.K8sClient.Status().Update(th.Ctx, keystone.DeepCopy())).Should(gomega.Succeed())
+
+	th.Logger.Info("KeystoneAPI created", "KeystoneAPI", name)
+	return name
+}
+
+// CreateKeystoneAPIWithFixture creates a KeystoneAPI CR and configures
+// its endpoints to point to the KeystoneAPIFixture that simulate the
+// keystone-api behavior.
+func (th *TestHelper) CreateKeystoneAPIWithFixture(
+	namespace string, fixture *apis.KeystoneAPIFixture,
```

Comment:
Should the fixture here also move into the API package?

Reply:
Good point. I did not caught it as KeystoneAPIFixture does not have any keystone-operator dependency. But logically that fixture should move too. I will respin